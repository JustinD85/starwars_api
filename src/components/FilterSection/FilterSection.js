import React from 'react';
import './FilterSection.scss';

const FilterSection = (props) => {
  const { setCurrentFilter, favoriteCount } = props
  return (
    <section className="FilterSection">
      <button onClick={() => setCurrentFilter('people')} className="people">people</button>
      <button onClick={() => setCurrentFilter('vehicles')} className="vehicle">vehicles</button>
      <button onClick={() => setCurrentFilter('planets')} className="planet">planets</button>
      <button className="favorite">Favorites: {favoriteCount}</button>
    </section>
  )
}

export default FilterSection;