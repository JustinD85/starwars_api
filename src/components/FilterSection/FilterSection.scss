.FilterSection {
  padding: 1%;
  height: 10%;
  width: 100%;
  display: grid;
  grid-template-areas:
    " vehicle people planet favorite";

  button {
    width: 100px;
    height: 50px;
    border: 1px solid red;
    margin: 1%;
    border-radius: 25px;
    background: #000;
    color: gold;
    text-transform: capitalize;
  }

  .people {
    grid-area: people;
  }

  .vehicle {
    grid-area: vehicle;
  }

  .planet {
    grid-area: planet;
  }

  .favorite {
    grid-area: favorite;
  }

}