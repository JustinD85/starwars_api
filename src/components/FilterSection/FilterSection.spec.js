import React from 'react';
import { shallow } from 'enzyme'
import FilterSection from './FilterSection'

describe('<FilterSection/>', () => {
  let wrapper, expected, mockSetCurrentFilter, mockFavCount
  beforeEach(() => {
    mockSetCurrentFilter = jest.fn()
    mockFavCount = 0
    wrapper = shallow(<FilterSection
      setCurrentFilter={mockSetCurrentFilter}
      favoriteCount={mockFavCount}
    />)
  })

  it('to Match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should call setCurrent filter when clicked', () => {
    //execution
    wrapper.find('.people').simulate('click')
    wrapper.find('.vehicle').simulate('click')
    wrapper.find('.planet').simulate('click')
    wrapper.find('.favorite').simulate('click')

    //expectation
    expect(mockSetCurrentFilter).toHaveBeenCalledTimes(4)
  })

  it('should show favorite count if non-zero', () => {
    //setup
    expected = "Favorites: 1"
    mockFavCount = 1
    wrapper = shallow(<FilterSection
      setCurrentFilter={mockSetCurrentFilter}
      favoriteCount={mockFavCount}
    />)

    //expectation
    expect(wrapper.find('.favorite').text()).toEqual(expected)
  })

  it('should show a message if no favorites', () => {
    //setup
    expected = "Favorites: none"
    //expectation
    expect(wrapper.find('.favorite').text()).toEqual(expected)
  })
})