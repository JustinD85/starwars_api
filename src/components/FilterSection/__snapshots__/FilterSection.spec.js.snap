// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterSection/> to Match snapshot 1`] = `
<section
  className="FilterSection"
>
  <button
    className="people"
    onClick={[Function]}
  >
    people
  </button>
  <button
    className="vehicle"
    onClick={[Function]}
  >
    vehicles
  </button>
  <button
    className="planet"
    onClick={[Function]}
  >
    planets
  </button>
  <button
    className="favorite"
    onClick={[Function]}
  >
    Favorites: 
    none
  </button>
</section>
`;
